editor/gui/scene_tree_editor.cpp

```diff
@@ -369,26 +369,26 @@ void SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
 					msg_temp += String::utf8("•  ") + String(E.name) + "\n";
 				}
 			}
-		}
-		if (num_connections >= 1 || num_groups >= 1) {
-			if (num_groups < 1) {
-				msg_temp += "\n";
-			}
-			msg_temp += TTR("Click to show signals dock.");
+		} else {
+			msg_temp += "\n";
```

Comment:
This adds the new line regardless, not sure that is correct, I'd just add the new line before adding the ending below, like so:
```c++
msg_temp += "\n" + msg_temp_end;
```

Reply:
I also don't know...  But if node doesn't have any groups we doesn't need to add `\n` because it will become unnecessary empty line. If we have groups we will get the `\n` along with `msg_temp += String::utf8("•  ") + String(E.name) + "\n";`